@here/harp-mapview/lib/ClipPlanesEvaluator.ts

```diff
@@ -588,27 +615,29 @@ export class TiltBasedClipPlanesEvaluator extends AltitudeBasedClipPlanesEvaluat
             // cos(90 - tilt) = farPlaneExt / frustumIntersectTop
             // farPlaneExt = cos(90 - tilt) * frustumIntersectTop
             const tiltAngleRestCos = Math.cos(Math.PI / 2 - cameraTilt);
-            clipPlanes.far += frustumGroundProj.top * tiltAngleRestCos;
+            viewRanges.far += frustumGroundProj.top * tiltAngleRestCos;
             // Similar formulas may be derived for near plane distance:
             // cos(90 - tilt) = nearPlaneExt / frustumIntersectBottom
             // nearPlaneExt = frustumIntersectBottom * cos(90 - tilt)
-            clipPlanes.near -= frustumGroundProj.bottom * tiltAngleRestCos;
+            viewRanges.near -= frustumGroundProj.bottom * tiltAngleRestCos;
         }
         // Clamp values to constraints.
         const farMax = lookAtDist * this.farMaxRatio;
-        clipPlanes.near = Math.max(clipPlanes.near, this.nearMin);
-        clipPlanes.far = Math.min(clipPlanes.far, farMax);
-        clipPlanes.maximum = farMax;
-        return clipPlanes;
+        viewRanges.near = Math.max(viewRanges.near - this.nearFarMargin / 2, this.nearMin);
+        viewRanges.far = Math.min(viewRanges.far, farMax);
+        viewRanges.far = Math.max(viewRanges.far, viewRanges.near + this.nearFarMargin);
+        viewRanges.minimum = this.nearMin;
+        viewRanges.maximum = farMax;
+        return viewRanges;
     }
 
     private evaluateSphericalProj(camera: THREE.Camera, projection: Projection): ViewRanges {
         assert(projection.type === ProjectionType.Spherical);
-        const clipPlanes = { ...this.minimumViewRange };
+        const viewRanges = { ...this.minimumViewRange };
 
         // Near plance calculus is pretty straightforward and does not depent on camera tilt:
         const cameraAltitude = this.getCameraAltitude(camera, projection);
-        clipPlanes.near = cameraAltitude - this.maxElevation;
+        viewRanges.near = cameraAltitude - this.maxElevation;
```

Comment:
Just as a note, despite being a good solution, we need to know what are the consequences of not pushing the nearPlane as aggresively as possible in our current setup.

This pushes the ``nearPlane`` to minimum values at street levels (or any height smaller than the Burj Khalifa), which might be nice to avoid nearPlaneClipping, but absolutely destroys the [depth buffer precision](https://developer.nvidia.com/content/depth-precision-visualized):

```
The graph above shows the “standard”, vanilla depth mapping used in D3D and similar APIs. You can immediately see how the 1/z curve leads to bunching up values close to the near plane, and the values close to the far plane are quite spread out.

It's also easy to see why the near plane has such a profound effect on depth precision. Pulling in the near plane will make the d range skyrocket up toward the asymptote of the 1/z curve, leading to an even more lopsided distribution of values.
```

Against this problem, we have 2 possible solutions:
* Push the min nearPlane to some distance that provides enough depth precision and fade fragment in a certain distance from the nearPlane (simple).
* Support a proper **ReversedDepth Projection matrix** which distributes the precision more evenly across the whole frustum, removing the precision dependency on the nearPlane. This one I like, but we need to get a few PRs into three to properly make it work...

Reply:
@amsXYZ, what do you mean that pushing the near plane to minimum values at street level destroys the depth buffer precision? If the near plane is as close as possible to the scene, then we have the best precision, to the near plane, i.e. close to the streets, which is what we want. So if you have a max elevation of 10 and min of 0, then the near -> far plane range would be 10, I don't see how you can calculate a depth range better than that.

Reply:
@amsXYZ I think I could try with first approach (near fading), the second one is bold (as you also know), but still I would opt to merge this stuff, cause it already fixes some issues. I am REALLY afraid of extending this PR to infinity, such I have with previous ticket, cause we already agreed to do things step by step, as long as they are not moving us back, and this one definitely moves us forward, just compare InterpolatedClipPlanesEvaluator (legacy code) with new approaches - easy to test on MapViewDemo app (play with MapControls->Frustum->Planes evaluator, enable Debug Camera, turn on terrain). You will quickly note the difference.

Reply:
@kkostecki, I agree, I think we should focus rather in getting this PR through than fixing these issues... @amsXYZ , maybe create some issues for this and they can be followed up?

Reply:
I think we have already agreed, that Burj Khalifa as our near constraint (distance limit) will be removed soon, of course I vote for @amsXYZ comments and believe that near fading seems to be most reasonable solution. I wrote about it in solutioning page too:
https://confluence.in.here.com/pages/viewpage.action?pageId=810200202